string.c

```diff
@@ -11525,6 +11525,178 @@ rb_enc_interned_str_cstr(const char *ptr, rb_encoding *enc)
  *  without a ``!'' return a new String.  However, there are
  *  exceptions, such as String#[]=.
  *
+ *  == What's Here
+ *
+ *  === Methods for Creating a \String
+ *  - ::new::
+ *  - ::try_convert::
+ *  === Methods for Setting \String State
+ *  - #+@::
+ *  - #-@::
+ *  - #freeze::
+ *  - #kconv::
+ *  === Methods for Querying
+ *  - #=~::
+ *  - #ascii_only?::
+ *  - #bytesize::
+ *  - #count::
+ *  - #empty?::
+ *  - #encoding::
+ *  - #end_with?::
+ *  - #eql?::
+ *  - #hash::
+ *  - #include?::
+ *  - #index::
+ *  - #iseuc::
+ *  - #isjis::
+ *  - #issjis::
+ *  - #isutf8::
+ *  - #length::
+ *  - #match::
+ *  - #match?::
+ *  - #rindex::
+ *  - #size::
+ *  - #start_with?::
+ *  - #sum::
+ *  - #unicode_normalized?::
+ *  - #valid_encoding?::
+ *  === Methods for Comparing
+ *  - #<=>::
+ *  - #==::
+ *  - #===::
+ *  - #casecmp::
+ *  - #casecmp?::
+ *  === Methods for Fetching
+ *  - #[]::
+ *  - #byteslice::
+ *  - #chr::
+ *  - #getbyte::
+ *  === Methods for Modifying
+ *  - #<<::
+ *  - #[]=::
+ *  - #capitalize!::
+ *  - #chomp!::
+ *  - #chop!::
+ *  - #downcase!::
+ *  - #encode!::
+ *  - #gsub!::
+ *  - #force_encoding::
+ *  - #initialize_copy::
+ *  - #insert::
+ *  - #lstrip!::
```

Comment:
`lstrip!`, `rstrip!`, `scrub!`, `slice!`, `chomp!`, and `chop!` should probably be moved to `Methods for Deleting`.

Reply:
Done.